### PR TITLE
PATCH call not proxied correctly to active node

### PR DESCRIPTION
**Pull request by meln1k**
Summary: added a workaround for http methods unsupported by java

JIRA issues: MARATHON-8095

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala [hunk 1]
```diff
@@ -105,7 +105,12 @@ class JavaUrlConnectionRequestForwarder(
   }
 
   private def copyRequestToConnection(leaderConnection: HttpURLConnection, request: HttpServletRequest): Try[Done] = Try {
-    leaderConnection.setRequestMethod(request.getMethod)
+    if (request.getMethod == "PATCH") { //workaround to overcome java restrictions, see https://bugs.openjdk.java.net/browse/JDK-7016595
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala]: nit: a space after `//`.

File: src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala [hunk 2]
```diff
@@ -105,7 +105,12 @@ class JavaUrlConnectionRequestForwarder(
...
+      leaderConnection.setRequestProperty("X-HTTP-Method-Override", "PATCH")
```
Comment (zen-dog) [inline src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala]: Does jetty automatically deal with `"X-HTTP-Method-Override", "PATCH"`? Did you try it locally?

File: src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala [hunk 3]
```diff
@@ -104,8 +104,18 @@ class JavaUrlConnectionRequestForwarder(
     leaderConnection.addRequestProperty(HEADER_FORWARDED_FOR, forwardedFor)
   }
 
+  private def setPatchMethod(connection: HttpURLConnection): Unit = {
+    val methodsField = classOf[HttpURLConnection].getDeclaredField("method")
+    methodsField.setAccessible(true)
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/api/forwarder/JavaUrlConnectionRequestForwarder.scala]: do we have to set it accessible every time? Or just once?

I think I'd like to see an object that encapsulates this unfortunate but necessary workaround

```
/**
 * Workaround to overcome Java restrictions; see https://bugs.openjdk.java.net/browse/JDK-7016595
 */
object HttpUrlConnectionWorkaround {
  val methodsField = classOf[HttpURLConnection].getDeclaredField("method")
  methodsField.setAccessible(true)

  def setMethod(connection: HttpURLConnection, method: String): Unit = {
    if (method == "PATCH")
      methodsField.set(connection, method)
    else
      connection.setRequestMethod(method)
  }
}
```

**Comment by timcharper** [comment]
This is just for the sync proxy in 1.6.x, which is deprecated and slated for removal in 1.8.x

**Pull request merged**